### PR TITLE
style: optimize steps style

### LINKED ISSUE #23565: steps style in compact theme
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/steps-cn/#components-steps-demo-vertical](https://ant.design/components/steps-cn/#components-steps-demo-vertical)

### Steps to reproduce
- 垂直方向上，紧凑模式使用了 small
- ant-steps-item-tail 距离边界 12px了
- RTL 模式下 未处理
### What is expected?
- 12px

### What is actually happening?
- 16px

| Environment | Info |
|---|---|
| antd | 4.1.4 |
| React | The latest |
| System | mac |
| Browser | chrome |

---

你在 compact 中处理了 ant-steps-item-tail 

image

我不太清楚你这个加在哪了，你直接 处理下吧 😂😂，我就不找了

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/23565

### 💡 Background and solution
- 优化 steps less

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |        none   |
| 🇨🇳 Chinese |  none         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/style/themes/compact.less [hunk 1]
```diff
@@ -164,8 +164,9 @@
 @steps-dot-top: 4px;
 @steps-icon-top: 0px;
 @steps-small-icon-size: 20px;
-@steps-vertical-icon-margin: 0 12px 0 0;
-
+@steps-vertical-icon-margin-width: 12px;
```
Comment (afc163) [inline components/style/themes/compact.less:167]: margin-width 语义不明。

File: components/style/themes/compact.less [hunk 2]
```diff
@@ -164,8 +164,9 @@
 @steps-dot-top: 4px;
 @steps-icon-top: 0px;
 @steps-small-icon-size: 20px;
-@steps-vertical-icon-margin: 0 12px 0 0;
```
Comment (afc163) [inline components/style/themes/compact.less:167]: 原有的变量最好不要删掉。

Comment (xrkffgg) [reply components/style/themes/compact.less:167]: 应该最开始 就是 一个的，他在改造的时候，弄成了 4个。

File: components/style/themes/default.less [hunk 3]
```diff
@@ -929,7 +929,9 @@
 @steps-current-dot-size: 10px;
 @steps-desciption-max-width: 140px;
 @steps-nav-content-max-width: auto;
-@steps-vertical-icon-margin: 0 16px 0 0;
+@steps-vertical-icon-width: 16px;
+@steps-vertical-tail-width: 16px;
+@steps-vertical-tail-width-small: 12px;
```
Comment (afc163) [inline components/style/themes/default.less:934]:
```suggestion
@steps-vertical-tail-width-sm: 12px;
```

**Pull request merged**